### ISSUE TITLE
Some Issues

### ISSUE THREAD

**Issue by HadesIzanami** [Ryochan7/DS4Windows#599]
The firts issue is when the battery is in 10% the DS4Windows show 0% and also the "flash at" funtion doesnt work. I dont know if this problem arose from using multiple profiles.

![captura](https://user-images.githubusercontent.com/42960801/53928312-44a88680-404f-11e9-8d94-a8f1ac7887b9.PNG)

And the second issue I found is when I was looking at the colors that can be changed in the "special actions", they can not be edited and it gives an error.

![captura2](https://user-images.githubusercontent.com/42960801/53928514-065f9700-4050-11e9-962c-350700e261c4.PNG)

**Comment by Ryochan7** [comment]
Flash and pulse work for me fine. Where are you getting the 10% battery level number from?

I have confirmed the second problem on my end. I will look into it.

**Comment by Ryochan7** [comment]
This seems to have broken with version 1.6.12. Looking at the old commits, I cannot see when the problem even happened. The only real factor was targetting a different version of the .NET Framework. I'm going to blame VS for this one.

**Comment by Ryochan7** [comment]
Found the offending commit. It was an issue introduced with the 360 steering wheel mode. Still going to blame VS for this.

1eed418022640fc8f77a22c844204b480b975100

**Comment by HadesIzanami** [comment]
> Flash and pulse work for me fine. Where are you getting the 10% battery level number from?
> 
> I have confirmed the second problem on my end. I will look into it.
> 
> Edit: sorry about closing. button slip up there.

Its a very rare case. I will try explain all. When I use "flash at" function, in 10% of  battery, doest work, I mean, when my DS4 is in 11%, still show it in the program, but, when this goes down to 10%, automatic the program shows 0% of battery, and "flash at" show a continue flash on, But, "Pulse" function works, and "flash at" only works when I use a diferente value, like 20% for example.

**Comment by Ryochan7** [comment]
The battery level returned from the controller doesn't cover a large range so the battery level cannot be granular. For a BT controller, the range is 0 - 8  and for USB its 0 - 11. Also, I looked at the underlying events and DS4Windows only sets the Flash/Pulse at value to the closest 10s value. In that case, setting Pulse At to 11% would actually cause the setting to round down to 10%. I didn't make that restriction and I don't know why it works like that.

**Comment by Ryochan7** [comment]
Just got my BT controller down to the 0% mark. Flash does not work but Pulse does. Now I have something to look into.

**Comment by Ryochan7** [comment]
Checked this against Jays2Kings DS4Windows version 1.4.52 and it works the same way. Looking at the source code, Flash At when the battery level is at 0% is designed to work the same as if the battery is fully charged. Inline comments state that it is actually the designed behavior. Weird decision.

```
else if (device.getBattery() <= getFlashAt(deviceNum) && getFlashType(deviceNum) == 0 && !defaultLight && !device.isCharging())
{
    int level = device.getBattery() / 10;
    if (level >= 10)
        level = 0; // all values of ~0% or >~100% are rendered the same

    haptics.LightBarFlashDurationOn = BatteryIndicatorDurations[level, 0];
    haptics.LightBarFlashDurationOff = BatteryIndicatorDurations[level, 1];
}
```

https://github.com/Ryochan7/DS4Windows/blob/jay/DS4Windows/DS4Control/DS4LightBar.cs#L297

**Comment by Ryochan7** [comment]
These issues have been fixed in version 1.6.13. Thank you for bringing up these problems.
